### ISSUE TITLE
Applying source does not update values in editor-form

### ISSUE THREAD

**Issue by miyconst** [Juicy/juicy-tile-editor#66]
Applying source by `Source` button updates tiles setup, but the values in the `juicy-tile-editor` form is not being updated.

Some refresh required.

**Comment by miyconst** [comment]
The issue was because `allItems` dictionary which is `Dictionary<id, tile>` was not updated on `juicy-tile-list.setup` change.

I took the update logic from `attributeChanged` method to `onSetupChanged` and call explicitly whenever a new json setup applied.

@tomalec take a look please.

**Comment by tomalec** [comment]
Have you tried conventional `setupChanged`?
Entire fix LGTM except the fact, that now we have native: `attributeChangedCallback`, polymers: `attributeChanged`, `setupChanged`, and adding another custom `onSetupChanged` method could provide more confusion.

Maybe we should use Polymer's `setupChanged` so it will be used automatically as a setter for setup. 

On the other hand maybe it's better not to rely on Polymer's 0.5 features as we will need to port it to Vanilla or 1.0

Maybe we could rename it then to `re-load`, so we will have such life-cycle methods:
- `created` (WC/Polymer) -  just prepares custom element itself
- `domReady` (WC/Polymer) -  "loads": setup, child items, starts listeners and observers to `refresh` the list,
- `detached` (WC/Polymer) - stops observers,
- `refresh` (ours) - (re-)`packItems` and renders child elements accordingly,
- `packItems` which just pack abstract items, and returns a result
  load,
- `reload` **NEW** - reads setup and children list once again, and `refresh`es

To sum up if conventional Polymer `setupChanged` works as expected I'd suggest to use it, and create an alias for `reload`, if it doesn't I would go with your solution renamed to `reload`.

**Comment by miyconst** [comment]
I could try using `setupChanged` of Polymer, but `attributeChangedCallback` does not fire when `this.setup` is changed like `this.editedTiles.setup = this.source`, that is why I add a function to be able to call it manually.

I will try to use `setupChanged` and if it does not work then rename `onSetupChanged` to `reload`.

**Comment by miyconst** [comment]
Polymer's `setupChanged` callback works as expected. Closing the issue.
